test/python/mitigation/generate_data.py

```diff
@@ -0,0 +1,294 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# pylint: disable=invalid-name
+
+"""Data generator for readout mitigation tests"""
+from numpy import array
+from qiskit import QiskitError
+
+try:
+    from qiskit import Aer
+    from qiskit.providers.aer import noise
+
+    HAS_AER = True
+except ImportError:
+    HAS_AER = False
+
+# For simulation
+import qiskit.utils.mitigation as mit
+from qiskit import QuantumRegister, QuantumCircuit
+
+SEED = 100
+SHOTS = 10000
+
+
+def execute_circs(qc_list, sim, noise_model=None):
```

Comment:
Do we really need to pre-generate test data? I hate these sort of tests since its inconvenient having to generate data everytime you change something, and these tests should not be slow to simulate.

Reply:
I don't object to removing this; but maybe in a different PR?

Reply:
Sure, doing this in followup PR is fine as long as it does eventually get done